File: C++/B/65.cpp
#include <iostream>
#include <vector>
using namespace std;
int main()
{
    int n, ans = 0;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++)
    {
        cin >> a[i];
    }
    int i = 1;
    while (i != 2 && ans < n)
    {
        ans++;
        i = a[i - 1];
    }
    if (ans < n)
        cout << ans << endl;
    else
        cout << "-1\n";
}

File: ruby/B/65.rb
n = gets.to_i
arr = []
b, ans = 1, 1
n.times do
  arr << gets.to_i
end
for i in 0..n - 1
  break if b == 2
  ans += 1
  b = arr[b - 1]
end
p ans < n + 1 ? ans - 1 : -1
